cli/dinghy/machine.rb

```diff
@@ -117,8 +118,16 @@ def system(*cmd)
     Kernel.system("docker-machine", *cmd)
   end
 
+  def preferences
+    @preferences ||= Preferences.load
+  end
+
   def machine_name
-    'dinghy'
+    if (defined?(preferences[:machine_name])).nil?
```

Comment:
ruby hashes return nil for non-existent keys, so wrapping this access in a call to `defined?` isn't necessary